source/PlayerInfo.cpp

```diff
@@ -1115,9 +1124,42 @@ void PlayerInfo::SellShip(const Ship *selected)
 				stock[it.first] += it.second;
 
 			accounts.AddCredits(cost);
+			ForgetShip(*it->get());
 			ships.erase(it);
 			flagship.reset();
-			return;
+			break;
+		}
+}
+
+
+
+// Take the ship from the player, if a model is specified this will permanently remove outfits in said model,
+// instead of allowing the player to buy them back, by putting them in the stock.
+void PlayerInfo::TakeShip(const Ship *shipToTake, const Ship *outfitsToDestroy)
+{
+	for(auto it = ships.begin(); it != ships.end(); ++it)
+		if(it->get() == shipToTake)
+		{
+			// Record the transfer of this ship in the depreciation and stock info.
+			stockDepreciation.Buy(*shipToTake, date.DaysSinceEpoch(), &depreciation);
+			for(const auto &it : shipToTake->Outfits())
+			{
+				// We only take all of the outfits specified in the model without putting them in the stock.
+				// The extra outfits of this ship are transfered into the stock.
+				if(outfitsToDestroy)
+				{
+					auto outfit = outfitsToDestroy->Outfits().find(it.first);
+					int amountRequired = (outfit != outfitsToDestroy->Outfits().end() ? outfit->second : 0);
+					if(amountRequired < it.second)
+						stock[it.first] += it.second - amountRequired;
+				}
+				else
+					stock[it.first] += it.second;
```

Comment:
Is it ok that this still seems to add the destroyed outfits to the stock depreciation?

Reply:
There is a with outfit keyword, just not sure if it works properly here I'll take a look tomorrow

Reply:
wait no there is no issue here it just adds them if there is no need for outfits or there are extra ones